Makefile

```diff
@@ -0,0 +1,21 @@
+#! /usr/bin/env make -f
+# This Makefile assumes an activated Python virtual environment.
+
+.PHONY:	all
+all:	dist
+
+dist:	dist/.dist
+dist/.dist:
+		python setup.py bdist
+		python setup.py sdist
+		touch dist/.dist
+
+.PHONY:	check
```

Comment:
Can we make this check a requirement for the `dist` step?

Reply:
We can, but that’s a little annoying: `check` doesn’t write a persistent output which mean it _always_ runs. If I add it as a dependency to `dist`
```Makefile
dist: dist/.dist
dist/.dist: check
	python setup.py bdist
	python setup.py sdist
	touch dist/.dist
```
then checks run whether a distribution artifact exists or not. Touching a phony file like we do with `dist/.dist` seems to add more clutter than is worth.

Reply:
Note that the Stripe folks also use a [Makefile](https://github.com/stripe/stripe-python/blob/master/Makefile)…

Reply:
> We can, but that’s a little annoying: `check` doesn’t write a persistent output which mean it _always_ runs. If I add it as a dependency to `dist`

I think we need to have different options for building and packaging. For the `build` , we could pass a `skipTests` flag to optionally turn it off when needed.

Also, I think we can use this `Makefile` to perform the checks and tests in various Actions workflows, instead of doing the checks as separate workflow jobs. That would let us avoid copying these checks, which can be error-prone and problematic :

https://github.com/jenstroeger/python-package-template/blob/7292ee9804ba6cae77b63726ae9691e9aeb154e4/.github/workflows/pull-request.yml#L29-L53

https://github.com/jenstroeger/python-package-template/blob/7292ee9804ba6cae77b63726ae9691e9aeb154e4/.github/workflows/check-and-release-main.yml#L10-L34

Reply:
Please see commit https://github.com/jenstroeger/python-package-template/pull/74/commits/139ef23b1619e8b00394129043c3c621018f54ed, and my [comment below](https://github.com/jenstroeger/python-package-template/pull/74#issuecomment-1031115562).